pkg/controller/installation/products/mobiledeveloperconsole/reconciler.go

```diff
@@ -0,0 +1,246 @@
+	config, err := configManager.ReadMobileDeveloperConsole()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read mobile developer console")
+	}
+
+	if config.GetNamespace() == "" {
+		config.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+
+	err = config.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "mdc config is not valid")
+	}
+
+	logger := logrus.WithFields(logrus.Fields{"product": config.GetProductName()})
+
+	return &Reconciler{
+		ConfigManager: configManager,
+		Config:        config,
+		mpm:           mpm,
+		logger:        logger,
+		Reconciler:    resources.NewReconciler(mpm),
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation, product *v1alpha1.InstallationProductStatus, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+
+	phase, err := r.ReconcileNamespace(ctx, r.Config.GetNamespace(), inst, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	version, err := resources.NewVersion(v1alpha1.OperatorVersionMDC)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "invalid version number for mdc")
+	}
+	phase, err = r.ReconcileSubscription(
+		ctx,
+		inst,
+		marketplace.Target{Pkg: defaultSubscriptionName, Channel: marketplace.IntegreatlyChannel, Namespace: r.Config.GetNamespace()},
+		client,
+		version,
+	)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileCustomResource(ctx, client, inst)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.handleProgress(ctx, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	product.Host = r.Config.GetHost()
+	product.Version = r.Config.GetProductVersion()
+
+	r.logger.Infof("%s has reconciled successfully", r.Config.GetProductName())
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) GetPreflightObject(ns string) runtime.Object {
+	return nil
+}
+
+func (r *Reconciler) reconcileCustomResource(ctx context.Context, client pkgclient.Client, inst *v1alpha1.Installation) (v1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling mobile-developer-console custom resource")
+
+	clientSecret, err := r.getOauthClientSecret(ctx, client)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	cr := &mdc.MobileDeveloperConsole{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      resourceName,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: mdc.SchemeGroupVersion.String(),
+			Kind:       "MobileDeveloperConsole",
+		},
+		Spec: mdc.MobileDeveloperConsoleSpec{
+			OAuthClientId:     inst.Spec.NamespacePrefix + string(r.Config.GetProductName()),
+			OAuthClientSecret: string(clientSecret),
+		},
+	}
+
+	// attempt to create the custom resource
+	if err := client.Create(ctx, cr); err != nil && !k8serr.IsAlreadyExists(err) {
```

Comment:
perhaps use controllerutil.CreateOrUpdate like [here](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/controller/installation/products/mobilesecurityservice/reconciler.go#L140-L144)

Reply:
yes, I think this should be changed